src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/CodeGenerator.cs

```diff
@@ -38,7 +38,7 @@ private static MethodInfo ObjectEquals
             {
                 if (s_objectEquals == null)
                 {
-                    s_objectEquals = Globals.TypeOfObject.GetMethod("Equals", BindingFlags.Public | BindingFlags.Static);
+                    s_objectEquals = typeof(object).GetMethod("Equals", BindingFlags.Public | BindingFlags.Static);
```

Comment:
I realize this isn't new, but could we specify the parameter types in the `GetMethod` call?

Reply:
We can do that in the `master` branch with this follow up issue: #42861.